### DISCUSSION TITLE
What's the best compression method?

### DISCUSSION THREAD

**Discussion by KingOfTNT** [miurahr/py7zr#561] [Q&A]
There are a lot of options for compressing and I don't know which one is the best.

Could you please help figure out the best option for compression? I care about compression ratio and not time.
I tried asking on Reddit and no one answered yet.

I'm not sure I should ask this here, so I'm sorry if I'm wrong.

**Comment by miurahr** [comment]
I don't know best. There are many methods because original 7zip specification has and py7zr try to be compatible with original 7-zip.

You can see an one of  benchmark results that was draw at a version in Apr. 2021. 
https://github.com/miurahr/py7zr/wiki/Benchmarks

and  you can see a ratio and a speed of some versions at https://github.com/miurahr/py7zr/issues/297 "Tracking benchmark results" issue.

**Comment by noisysoil** [comment]
> There are a lot of options for compressing and I don't know which one is the best.
> 
> Could you please help figure out the best option for compression? I care about compression ratio and not time. I tried asking on Reddit and no one answered yet.
> 
> I'm not sure I should ask this here, so I'm sorry if I'm wrong.

@KingOfTNT 

[{"id": py7zr.FILTER_LZMA2, "preset": 9}]

This is after much testing in terms of compression ratio / time / versatility - as in OS support - vs others (including zstd).

**Comment by chirag127** [comment]
what is default?

**Comment by hiroshil** [comment]
> what is default?

ZStandard. if you use password it is 7zAES
